api/coupons.go

```diff
@@ -43,3 +43,22 @@ func (a *API) CouponView(w http.ResponseWriter, r *http.Request) error {
 
 	return sendJSON(w, http.StatusOK, coupon)
 }
+
+// CouponList returns all the coupons for the site. Requires admin permissions
+func (a *API) CouponList(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	log := getLogEntry(r)
+
+	couponCache := gcontext.GetCoupons(ctx)
+	if couponCache == nil {
+		return notFoundError("No coupons available")
```

Comment:
should we return a 404 here? Or just an empty list?

Reply:
an empty list might be better to handle in the frontend.